### PR TITLE
refactor(redis): switch to redis-py

### PR DESCRIPTION
**Pull request by sileht**
The change also makes mypy discovers some un-catched typing issue.

### CODE REVIEW

**Review by jd — COMMENTED** [review]
Did you try ddtrace?

**Comment by sileht** [comment]
> Did you try ddtrace?

nop, I will propose a patch

File: mergify_engine/migrations/__init__.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+
```
Comment (Syffe) [inline mergify_engine/migrations/__init__.py:15]:
```suggestion
```

File: mergify_engine/tests/functional/test_web_cli.py [hunk 2]
```diff
@@ -15,6 +15,8 @@
 # under the License.
 
 
+import asyncio
+
```
Comment (Syffe) [inline mergify_engine/tests/functional/test_web_cli.py:19]:
```suggestion
```

File: mergify_engine/service.py [hunk 3]
```diff
@@ -22,8 +22,13 @@
 from mergify_engine import logs
 
 
+global SERVICE_NAME
```
Comment (jd) [inline mergify_engine/service.py:25]:
```suggestion
```
does not do anything here

File: mergify_engine/service.py [hunk 4]
```diff
@@ -22,8 +22,13 @@
...
+SERVICE_NAME: str = "engine-<unknown>"
+
+
 def setup(service_name: str, dump_config: bool = True) -> None:
-    service_name = "engine-" + service_name
+    global SERVICE_NAME
+    SERVICE_NAME = "engine-" + service_name
```
Comment (jd) [inline mergify_engine/service.py:31]: why change the global variable? Is this used anything else?

File: mergify_engine/service.py [hunk 5]
```diff
@@ -40,8 +45,8 @@ def setup(service_name: str, dump_config: bool = True) -> None:
         sentry_sdk.utils.MAX_STRING_LENGTH = 2048
 
     ddtrace.config.version = _version
-    statsd.constant_tags.append(f"service:{service_name}")
-    ddtrace.config.service = service_name
+    statsd.constant_tags.append(f"service:{SERVICE_NAME}")
+    ddtrace.config.service = SERVICE_NAME
```
Comment (jd) [inline mergify_engine/service.py:49]: this whole sounds unrelated to this PR?

**Review by jd — DISMISSED** [review]

File: mergify_engine/service.py (hunk 4 above)
Comment (sileht) [reply mergify_engine/service.py:31]: This is used to set the redis client name with the service name.

File: mergify_engine/service.py (hunk 5 above)
Comment (sileht) [reply mergify_engine/service.py:48]: redis-py allow use to set the client name again, so I change that.

File: setup.cfg [hunk 6]
```diff
@@ -151,3 +154,6 @@ disallow_untyped_decorators = true
 show_error_codes = true
 disallow_untyped_calls = true
 plugins = pydantic.mypy
+
+[mypy-redis.*]
+ignore_errors = true
```
Comment (jd) [inline setup.cfg:159]: is this related to `type-redis` not being installed?
